core/src/sigverify.rs

```diff
@@ -33,6 +33,54 @@ pub type TxOffset = PinnedVec<u32>;
 
 type TxOffsets = (TxOffset, TxOffset, TxOffset, TxOffset, Vec<Vec<u32>>);
 
+#[derive(Debug, PartialEq, Eq)]
+struct PacketOffsets {
+    pub sig_len: u32,
+    pub sig_start: u32,
+    pub msg_start: u32,
+    pub pubkey_start: u32,
+}
+
+impl PacketOffsets {
+    pub fn new(sig_len: u32, sig_start: u32, msg_start: u32, pubkey_start: u32) -> Self {
+        Self {
+            sig_len,
+            sig_start,
+            msg_start,
+            pubkey_start,
+        }
+    }
+
+    pub fn new_with_packet(packet_offsets: PacketOffsets) -> Self {
```

Comment:
You don't need this method. See suggestion where you use it below

Reply:
done at b7f518d44fa0a4f3c33d3284ade4eae91f6190f7